### PR TITLE
CPUFreq shenanigans on Kobo

### PR DESCRIPTION
**Pull request by NiLuJe**
Wild-ass experiment switching to conservative instead of userspace (because, well, there's no userspace anymore :D).

Note that my H2O seems to grandly ignore all of that crap and switch freq all on its own anyway, so this may be a giant NOP.

### CODE REVIEW

**Comment by NiLuJe** [comment]
(As a counterpoint: my PW2 seems to actually honor CPUFreq, on the other hand).

**Comment by NiLuJe** [comment]
Fun fact: it also has a different frequency ladder:

```
996000 70845
396000 424702
```

vs.

```
166666 305779
500000 1745
1000000 96020
```

**Comment by NiLuJe** [comment]
TL;DR: Yet another spot where Kobo's kernel sucks ass.

**Comment by Frenzie** [comment]
Doesn't ondemand make more sense for our kind of loads (i.e., page turning once in a while)?

**Comment by NiLuJe** [comment]
I was wondering that myself, but I guess I'll have to test on a PW2 to see how that behaves... -_-"

**Comment by NiLuJe** [comment]
Fun fact: the PW2 is surprisingly decent in powersvae mode, stuck @400MHz. Library browsing is a bit more sluggish, but things are pretty okay in the reader :).

**Comment by NiLuJe** [comment]
That said, I forgot something obvious when comparing Oranges to Apples:

The H2O is still running on an A8 (I.MX 50), was the PW2 was already running on an A9 (i.MX 6SL)...

**Comment by NiLuJe** [comment]
Sigh. Now I remember why I went with conservative...

```
cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_available_governors
conservative powersave userspace performance
```

xD

**Comment by NiLuJe** [comment]
Okay, after a restart, at least `cpuinfo_cur_freq` stopped lying to me ;).

And now I know why we're murdering the battery: depending on how/when we're started, we might end up stuck on whatever step "userspace" left us in, and since we kill the "userspace"...

If we're stuck at 1GHz, good bye battery...

**Comment by NiLuJe** [comment]
The bad news is: yeaaah, conservative really doesn't work for us, it has a kind of rubber banding effect on the frequency that the eInk controller really doesn't like.

**Comment by NiLuJe** [comment]
And now I know why the PW2 doesn't step lower than 400MHz: that 166MHz step is the pits.

**Comment by NiLuJe** [comment]
TL;DR: Not sure what to do...

166MHz is basically unusable, 500MHz is terrible, and 1GHz murders the battery. Yay?

I'm inclined to switch to ondemand if available, and performance otherwise, and to hell with battery life?
Implementing userspace on our end is... impractical, we're not designed to do it properly, it's likely to be as terrible as conservative, and it's going to be platform (as-in device generation) specific anyway, so, hell.

**Comment by NiLuJe** [comment]
Okay, 500MHz is not completely awful, it's pretty okay in the reader, and mildly sluggish everywhere else.

**Comment by NiLuJe** [comment]
... Was trying to see if we couldn't simply re-use whatever the hell is driving userspace... and now it's not budging from the 1G step in Nickel, while it was nicely stepping up and down w/ nickel dead before a reboot (because there's still a nickel watchdog in one of those daemons, apparently :D).

*sigh*

**Comment by NiLuJe** [comment]
And an suspend/resume switch made it behave again.

*sigh*

**Comment by NiLuJe** [comment]
And by "behave", I mean, it's in userspace mode, there's no userspace left, and it switches frequency on its own. >_<".

**Comment by NiLuJe** [comment]
Meh, going to essentially revert this, and investigate again when the next FW comes out, because I'm living on he edge, 4.9 hasn't officially been deployed to Mk5, it's there, but not :D.

**Comment by NiLuJe** [comment]
@poire-z : Huh, that's good news, at least, that means my semi-revert will do the right thing for you ;).

**Comment by NiLuJe** [comment]
@poire-z : The GloHD is, what, Mark6? That looks surprisingly like my PW2 on an IMX6SL... ;).

**Comment by NiLuJe** [comment]
Interactive is essentially ondemand on steroids, so at least that makes some kind of sense as a default ;).

**Comment by poire-z** [comment]
```
# cat /proc/cpuinfo
Processor : ARMv7 Processor rev 10 (v7l)
BogoMIPS  : 1988.29
Features  : swp half thumb fastmult vfp edsp neon vfpv3
CPU implementer : 0x41
CPU architecture: 7
CPU variant  : 0x2
CPU part  : 0xc09
CPU revision : 10

Hardware  : Freescale i.MX 6SoloLite NTX Board
Revision  : 60200
```

> that means my semi-revert will do the right thing for you 

But your https://github.com/NiLuJe/koreader/commit/0f1773f7c8daf7e58059eb91cd5c620d89d153c8 would be switching me from _ondemand on steroids_ to _ondemand plain_ no ? Which is best? steroids or not ? :)

**Comment by NiLuJe** [comment]
@poire-z : That's indeed (roughly) the same CPU, so everything makes sense, yay!

Interactive is hard to tune right, even when you know what you're doing, so I have my doubts as to it being actually useful here, so, probably ondemand ;).

I don't even actually see the interactive tunable knobs in your dump, and I'm fairly sure they should be right there in that folder... >_<".

**Comment by NiLuJe** [comment]
Might depend on how old the kernel actually is... interactive began to be *really* interesting when it started doing it's input boost thingy (boosting the cpu on touchscreen events)...

**Comment by NiLuJe** [comment]
I can probably hide this being another !ondemand && !interactive test, just to be safe...

**Comment by NiLuJe** [comment]
Yeah, that's probably better.

At least I figured out why my battery was melting, so, there's that ;).

`cat /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq                                                                                                                                                        
1000000`

**Comment by NiLuJe** [comment]
Anyway, thanks for the data @poire-z , that's at least hope that things get somewhat better which each HW revision ;).
